src/analysis/inner_product_space/area_form.lean

```diff
@@ -0,0 +1,440 @@
+/-
+Copyright (c) 2022 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import analysis.inner_product_space.dual
+import analysis.inner_product_space.orientation
+import tactic.linear_combination
+
+/-!
+# Oriented two-dimensional real inner product spaces
+
+This file defines constructions specific to the geometry of an oriented two-dimensional real inner
+product space `E`.
+
+## Main declarations
+
+* `orientation.area_form`: an antisymmetric bilinear form `E →ₗ[ℝ] E →ₗ[ℝ] ℝ` (usual notation `ω`).
+  Morally, when `ω` is evaluated on two vectors, it gives the oriented area of the parallelogram
+  they span. (But mathlib does not yet have a construction of oriented area, and in fact the
+  construction of oriented area should pass through `ω`.)
+
+* `orientation.right_angle_rotation`: an isometric automorphism `E ≃ₗᵢ[ℝ] E` (usual notation `J`).
+  This automorphism squares to -1.  TODO: this agrees with the rotation by 90 degrees defined in
+  `orientation.rotation`.
+
+* `orientation.basis_right_angle_rotation`: for a nonzero vector `x` in `E`, the basis `![x, J x]`
+  for `E`.
+
+* `orientation.kahler`: a complex-valued real-bilinear map `E →ₗ[ℝ] E →ₗ[ℝ] ℂ`. Its real part is the
+  inner product and its imaginary part is `orientation.area_form`.  For vectors `x` and `y` in `E`,
+  the complex number `o.kahler x y` has modulus `∥x∥ * ∥y∥`. TODO: the argument of `o.kahler x y` is
+  the oriented angle (`orientation.oangle`) from `x` to `y`.
+
+## Implementation notes
+
+Notation `ω` for `orientation.area_form` and `J` for `orientation.right_angle_rotation` should be
+defined locally in each file which uses them, since otherwise one would need a more cumbersome
+notation which mentions the orientation explicitly (something like `ω[o]`).  Write
+
+```
+local notation `ω` := o.area_form
+local notation `J` := o.right_angle_rotation
+```
+
+-/
+
+noncomputable theory
+
+open_locale real_inner_product_space complex_conjugate
+open finite_dimensional
+
+local attribute [instance] fact_finite_dimensional_of_finrank_eq_succ
+
+variables {E : Type*} [inner_product_space ℝ E] [fact (finrank ℝ E = 2)]
+  (o : orientation ℝ E (fin 2))
+
+namespace orientation
+
+include o
+
+/-- An antisymmetric bilinear form on an oriented real inner product space of dimension 2 (usual
+notation `ω`).  When evaluated on two vectors, it gives the oriented area of the parallelogram they
+span. -/
+def area_form : E →ₗ[ℝ] E →ₗ[ℝ] ℝ :=
```

Comment:
Please make it irreducible right away, for Lean 4 transition. Same thing several times below.